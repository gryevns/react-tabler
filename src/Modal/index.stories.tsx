import React from "react";

import Button from "../Button";
import Modal from "../Modal";

export default {
    title: "Modal",
};

export const Active = () => {
    const [show, setShow] = React.useState(true);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
    return (
        <div
            style={{
                backgroundImage: "url(https://picsum.photos/id/128/1200)",
                height: "100vh",
            }}
        >
            <Button variant="primary" onClick={handleShow}>
                Show Modal
            </Button>
            <Modal show={show} onHide={handleClose}>
                <Modal.Header>Modal header</Modal.Header>
                <Modal.Body>
                    <p>Modal body text goes here.</p>
                </Modal.Body>
                <Modal.Footer>
                    <Button variant="secondary" onClick={handleClose}>
                        Close
                    </Button>
                    <Button variant="primary" onClick={handleClose}>
                        Save changes
                    </Button>
                </Modal.Footer>
            </Modal>
        </div>
    );
};
