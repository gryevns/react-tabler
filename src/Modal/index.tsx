import React from "react";
import cn from "classnames";

const ModalContext = React.createContext({
    onHide: () => {},
});

const ModalHeader = ({ children }) => {
    const { onHide } = React.useContext(ModalContext);
    return (
        <div className="modal-header">
            <div className="modal-title h4">{children}</div>
            <button type="button" className="close" onClick={onHide}>
                <span aria-hidden="true">&times;</span>
                <span className="sr-only">Close</span>
            </button>
        </div>
    );
};

const ModalBody = ({ children }) => <div className="modal-body">{children}</div>;

const ModalFooter = ({ children }) => <div className="modal-footer">{children}</div>;

export interface ModalProps {
    show: boolean;
    blur?: boolean;
    onHide: () => any;
    children: React.ReactNode;
}

const Modal = ({ show, blur = true, onHide, children }: ModalProps) => {
    if (!show) {
        return null;
    }
    const classes = cn("modal", { "modal-blur": blur });
    return (
        <ModalContext.Provider value={{ onHide }}>
            <div
                className={classes}
                role="dialog"
                aria-modal={true}
                style={{ display: "block" }}
            >
                <div className="modal-dialog" role="document">
                    <div className="modal-content">{children}</div>
                </div>
            </div>
        </ModalContext.Provider>
    );
};

Modal.Header = ModalHeader;
Modal.Body = ModalBody;
Modal.Footer = ModalFooter;

export default Modal;
