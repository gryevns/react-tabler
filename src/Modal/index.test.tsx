import React from "react";
import { mount } from "enzyme";

import Modal from "./index";

describe("Modal", () => {
    it("should match snapshot", () => {
        const handleClose = jest.fn();
        const wrapper = mount(
            <Modal show={true} onHide={handleClose}>
                <Modal.Header>Modal header</Modal.Header>
                <Modal.Body>
                    <p>Modal body text goes here.</p>
                </Modal.Body>
            </Modal>,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(".modal-dialog")).toHaveLength(1);
        wrapper.find("button.close").simulate("click");
        expect(handleClose).toHaveBeenCalled();
    });
    // TODO
});
