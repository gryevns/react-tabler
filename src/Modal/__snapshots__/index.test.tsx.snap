// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match size snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "onHide": null,
    }
  }
>
  <div
    aria-modal={true}
    className="modal modal-blur"
    role="dialog"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="modal-dialog modal-sm"
      role="document"
    >
      <div
        className="modal-content"
      >
        test
      </div>
    </div>
  </div>
</ContextProvider>
`;

exports[`Modal should match size snapshot 2`] = `
<ContextProvider
  value={
    Object {
      "onHide": null,
    }
  }
>
  <div
    aria-modal={true}
    className="modal modal-blur"
    role="dialog"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="modal-dialog modal-md"
      role="document"
    >
      <div
        className="modal-content"
      >
        test
      </div>
    </div>
  </div>
</ContextProvider>
`;

exports[`Modal should match size snapshot 3`] = `
<ContextProvider
  value={
    Object {
      "onHide": null,
    }
  }
>
  <div
    aria-modal={true}
    className="modal modal-blur"
    role="dialog"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="modal-dialog modal-lg"
      role="document"
    >
      <div
        className="modal-content"
      >
        test
      </div>
    </div>
  </div>
</ContextProvider>
`;

exports[`Modal should match snapshot 1`] = `
<Modal
  onHide={[MockFunction]}
  show={true}
>
  <div
    aria-modal={true}
    className="modal modal-blur"
    role="dialog"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <ModalHeader>
          <div
            className="modal-header"
          >
            <div
              className="modal-title h4"
            >
              Modal header
            </div>
            <button
              className="close"
              onClick={[MockFunction]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
              <span
                className="sr-only"
              >
                Close
              </span>
            </button>
          </div>
        </ModalHeader>
        <ModalBody>
          <div
            className="modal-body"
          >
            <p>
              Modal body text goes here.
            </p>
          </div>
        </ModalBody>
      </div>
    </div>
  </div>
</Modal>
`;
